types/node/module.d.ts

```diff
@@ -1,3 +1,29 @@
 declare module "module" {
-    export = NodeJS.Module;
+    class Module {
+        static runMain(): void;
+        static wrap(code: string): string;
+
+        /**
+         * @deprecated Deprecated since: v12.2.0. Please use createRequire() instead.
+         */
+        static createRequireFromPath(path: string): NodeJS.RequireFunction;
+        static createRequire(path: string): NodeJS.RequireFunction;
```

Comment:
A while ago I tried also to move Module from gobal to here but failed for some reason and never created a PR. Maybe because of some typescript version no longer supported...
Anyhow, could you please add `URL` support here: `static createRequire(path: string | URL)` and `import { URL } from "url";`.

That time I had also some you could reuse now: 
```
import m = require('module');
import { createRequireFromPath, createRequire } from 'module';
import { URL } from "url";

let rf: (m: string) => any;

rf = createRequire('mod');
rf = createRequireFromPath('mod');
rf = createRequire(new URL('file:///C:/path/'));

const aModule: NodeModule = new m.Module("s");
const bModule: NodeModule = new m("b", aModule);

const builtIn: string[] = m.builtinModules;
```

Reply:
Sure.